From 4cc4220f019923eddb979154e5ae4aec7324826b Mon Sep 17 00:00:00 2001
From: devi priya <quic_devipriy@quicinc.com>
Date: Sat, 11 Jun 2022 19:05:30 +0530
Subject: [PATCH 2/2] eph epv scaling factor update

Considering 2 sigma as the scaling factor for eph and epv fields
The sirf driver returns (eph/100)*2 and (epv/100)*2

Signed-off-by: devi priya <quic_devipriy@quicinc.com>
Index: gpsd-3.25/drivers/driver_sirf.c
===================================================================
--- gpsd-3.25.orig/drivers/driver_sirf.c
+++ gpsd-3.25/drivers/driver_sirf.c
@@ -839,10 +839,10 @@ static gps_mask_t sirf_msg_67_1(struct g
     session->newdata.epd = getbeu16(buf, 82) / 100.0;
     // distance traveled error in m * 100
     distance_travel_error = getbeu16(buf, 84) / 100.0;
-
-    session->newdata.eph = getbeu32(buf, 86) / 100.0;  /* Estimated horizontal position error * 100 */
+    /* Considering 2 sigma as the scaling factor for eph and epv fields */
+    session->newdata.eph = (getbeu32(buf, 86) / 100.0) * 2;  /* Estimated horizontal position error * 100 */
     // Estimated vertical position error * 100
-    session->newdata.epv = getbeu32(buf, 90) / 100.0;
+    session->newdata.epv = (getbeu32(buf, 90) / 100.0) * 2;
     // Estimated horizontal velocity error * 100
     session->newdata.eps = getbeu16(buf, 94) / 100.0;
     mask |= SPEEDERR_SET;
